src/render.jl

```diff
@@ -98,6 +98,52 @@ function url(m::Meta{:source})
     end
 end
 
+## Utilities
+# Return the index of the last non-empty string value in "A" (determined by `!isempty(A[i])`).
+function findlast{S <: AbstractString}(A::Array{S,1})
+    i = length(A)
+    while i > 0
+        isempty(A[i]) || break
+        i -= 1
+    end
+    return i
+end
```

Comment:
There's an implementation of this in `0.4`, called `findlast` as well. I'd suggest using that instead even though your version works just as well. Means we can discard code once everyone has moved to `0.4`/`0.5`. Perhaps something like:

```julia
if VERSION < v"0.4-"
    # returns the index of the previous element for which the function returns true, or zero if it never does
    function findprev(testf::Function, A, start)
        for i = start:-1:1
            testf(A[i]) && return i
        end
        0
    end
    findlast(testf::Function, A) = findprev(testf, A, length(A))
end
```

and the calling `findlast(x -> !isempty(x), path_arr)` where needed. This probably won't be as efficient as your one, but I doubt we've going to run into terrible slow downs here?